### PR TITLE
Added test with V4L CAP_PROP_CONVERT_RGB=false.

### PR DESCRIPTION
**Pull request by asmorkalov**
Test for https://github.com/opencv/opencv/pull/24180

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/videoio/test/test_camera.cpp [hunk 1]
```diff
@@ -218,6 +218,32 @@ TEST(DISABLED_videoio_camera, v4l_read_framesize)
...
+    std::cout << "New pixel format: " << capture.get(cv::CAP_PROP_FORMAT) << std::endl;
+
+    cv::Mat frame;
+    for (int i = 0; i < 10; i++)
+    {
+        int pixel_type  = capture.get(cv::CAP_PROP_FORMAT);
+        int channels    = CV_MAT_CN(pixel_type);
+        int pixel_bytes = CV_ELEM_SIZE(pixel_type);
+
+        // YUYV is expected for most of popular USB cam (COLOR_YUV2BGR_YUYV conversion)
```
Comment (mshabunin) [inline modules/videoio/test/test_camera.cpp:240]: I think this is not correct assumption, e.g on my laptop (pretty regular) I have 3 channels and 3 bytes and the tests fails.

**Review by mshabunin — COMMENTED** [review]
Can this test be moved to the test_v4l2.cpp (which uses vivid device)? It also turns CONVERT_BGR off and checks YUYV format, maybe  it should also force change image size to match original scenario?

File: modules/videoio/test/test_camera.cpp [hunk 2]
```diff
@@ -218,6 +218,32 @@ TEST(DISABLED_videoio_camera, v4l_read_framesize)
...
 
+TEST(DISABLED_videoio_camera, v4l_rgb_convert)
+{
+    VideoCapture capture(CAP_V4L2);
+    ASSERT_TRUE(capture.isOpened());
+    std::cout << "Camera 0 via " << capture.getBackendName() << " backend" << std::endl;
+    std::cout << " Frame width: " << capture.get(CAP_PROP_FRAME_WIDTH) << std::endl;
+    std::cout << "      height: " << capture.get(CAP_PROP_FRAME_HEIGHT) << std::endl;
+    std::cout << "Pixel format: " << capture.get(cv::CAP_PROP_FORMAT) << std::endl;
+
```
Comment (mshabunin) [inline modules/videoio/test/test_camera.cpp:229]: Lets check if camera uses YUYV by default:
```suggestion
    if (capture.get(CAP_PROP_FOURCC) != VideoWriter::fourcc('Y', 'U', 'Y', 'V'))
    {
        throw SkipTestException("Camera does not support YUYV format");
    }
```

**Comment by mshabunin** [comment]
There is a warning on Windows:
```
C:\build\precommit_windows64\4.x\opencv\modules\videoio\test\test_camera.cpp(241): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data [C:\build\precommit_windows64\build\modules\videoio\opencv_test_videoio.vcxproj]
```

**Comment by asmorkalov** [comment]
@mshabunin I fixed warning and applied your suggestion.  Could you take a look the PR and RISC-V CI failure. Looks like it's not related to the PR.

**Review by mshabunin — APPROVED** [review]
Fix for failed RISC-V workflows is here: https://github.com/opencv/ci-gha-workflow/pull/116

I think we can ignore CI failures and merge this PR.

**Pull request merged**